### PR TITLE
Tests ordering

### PR DESCRIPTION
**Pull request by milo**
- bug fix? no
- new feature? yes
- BC break? no, but temporary directory is required
- doc PR: will

Runner remebers last run tests state and runs failed and new tests as first.

### CODE REVIEW

File: src/Runner/Runner.php [hunk 1]
```diff
@@ -95,6 +101,27 @@ public function run()
 		}
 		$this->jobCount = count($this->jobs) + array_sum($this->results);
 
+		if ($this->tempDir) {
+			$this->orderFile = $this->tempDir . DIRECTORY_SEPARATOR . 'Tester.Runner-' . substr(md5(__FILE__), 0, 5) . '.list';
+			touch($this->orderFile);
```
Comment (dg) [inline src/Runner/Runner.php]: I usually run globally installed Runner, so maybe `getcwd()` should be in hash too, what you think?

**Comment by dg** [comment]
GREAT!

File: src/Runner/Runner.php (hunk 1 above)
Comment (milo) [inline src/Runner/Runner.php]: Probably yes. Will see how it will work in practice. Can by changed later to global, not hashing at all.

Comment (dg) [inline src/Runner/Runner.php]: This is actually true, there may not be hash.

File: src/Runner/Runner.php [hunk 2]
```diff
@@ -21,6 +21,9 @@ class Runner
 	/** @var int  run in parallel threads */
 	public $threadCount = 1;
 
+	/** @var string */
```
Comment (JanTvrdik) [inline src/Runner/Runner.php]: `NULL|string`?

File: src/Runner/Runner.php [hunk 3]
```diff
@@ -48,6 +51,9 @@ class Runner
 	/** @var bool */
 	private $interrupted;
 
+	/** @var string */
```
Comment (JanTvrdik) [inline src/Runner/Runner.php]: `NULL|string`

**Comment by milo** [comment]
@dg @JanTvrdik What about such approach? (have to fix tests yet)

**Comment by JanTvrdik** [comment]
@milo Looks awesome.

**Comment by dg** [comment]
Separated files for each test instead of one file per tested folder? I am not sure what is better, but current code is more readable.

**Comment by milo** [comment]
With separated files is much easier to update result of one test when change.

Reading 1 byte from 1000 files took 40ms on old server with rotation disks. I think performace will be OK.

**Pull request merged**